File: app/src/main/java/com/cradle/neptune/view/PdfViewActivity.java
package com.cradle.neptune.view;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.cradle.neptune.R;
import com.github.barteksc.pdfviewer.PDFView;

public class PdfViewActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pdf_view);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setTitle("Educational Poster");
        }
        Intent intent = getIntent();
        if (intent != null) {
            String filename = intent.getStringExtra("poster");
            setupPDFview(filename);
        }
    }

    private void setupPDFview(String filename) {
        PDFView pdfDocument = findViewById(R.id.pdfView);
        pdfDocument.fromAsset(filename).enableAntialiasing(true).spacing(8).load();

    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return super.onSupportNavigateUp();
    }
}

File: app/src/main/java/com/cradle/neptune/view/PdfViewActivity.kt
package com.cradle.neptune.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.cradle.neptune.R
import com.github.barteksc.pdfviewer.PDFView

class PdfViewActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pdf_view)
        if (supportActionBar != null) {
            supportActionBar?.setDisplayHomeAsUpEnabled(true)
            supportActionBar?.title = "Educational Poster"
        }
        intent?.getStringExtra("poster")?.also {
            setupPdfView(it)
        }
    }

    private fun setupPdfView(filename: String) {
        val pdfDocument = findViewById<PDFView>(R.id.pdfView)
        pdfDocument.fromAsset(filename).enableAntialiasing(true).spacing(8).load()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return super.onSupportNavigateUp()
    }
}